darity.go

```diff
@@ -91,6 +101,27 @@ func (c *Client) APIVersion() (int, error) {
 	return c.ioctl(c.kvm.Fd(), kvmGetAPIVersion, 0)
 }
 
+// CreateVM returns a VM struct built around the fd provided
+// by kvm.
+func (c *Client) CreateVM(t MachineType) (*VM, error) {
+	v, err := c.ioctl(c.kvm.Fd(), kvmCreateVM, 0)
+	if err != nil {
+		return nil, err
+	}
+	return &VM{
+		fd:    v,
+		ioctl: c.ioctl,
+	}, nil
+}
+
+type VM struct {
+	// File descriptor of the created VM
+	fd int
+
+	// ioctl syscall implementation
+	ioctl ioctlFunc
```

Comment:
I thought about that but the VM won't use the KVM device directly. It'll operate on it's own fd, so the `*Client` would be there only for the `vm.c.ioctl()` call. Maybe an `Ioctler` interface or something would save some duplication but we'd still need to pass the new fd.